packages/command-palette/src/use-command-palette.tsx

```diff
@@ -87,74 +104,80 @@ const useSiteToAction = () => {
 				search,
 			}: SiteToActionParameters[ 'properties' ]
 		) =>
-			( site: SiteExcerptData ): Command => {
+			( site: SiteData ): Command => {
 				const siteName = site.name || site.URL; // Use site.name if present, otherwise default to site.URL
 				return {
 					name: `${ site.ID }`,
 					label: `${ siteName }`,
 					subLabel: `${ site.URL }`,
 					searchLabel: `${ site.ID } ${ siteName } ${ site.URL }`,
 					callback: ( { close } ) => {
-						dispatch(
-							recordTracksEvent( 'calypso_hosting_command_palette_site_select', {
-								command: selectedCommand.name,
-								list_count: filteredSitesLength,
-								list_visible_count: listVisibleCount,
-								current_route: currentRoute,
-								search_text: search,
-								command_site_id: site.ID,
-								command_site_has_custom_domain: isCustomDomain( site.slug ),
-								command_site_plan_id: site.plan?.product_id,
-							} )
-						);
+						recordTracksEvent( 'calypso_hosting_command_palette_site_select', {
+							command: selectedCommand.name,
+							list_count: filteredSitesLength,
+							list_visible_count: listVisibleCount,
+							current_route: currentRoute,
+							search_text: search,
+							command_site_id: site.ID,
+							command_site_has_custom_domain: isCustomDomain( site.slug ),
+							command_site_plan_id: site.plan?.product_id,
+						} );
 						onClickSite( { site, close, command: selectedCommand } );
 					},
 					image: (
 						<FillDefaultIconWhite>
-							<SiteIcon site={ site } size={ 32 } />
+							{ site.icon?.img ? (
+								<SiteIcon src={ site.icon.img } alt="" />
+							) : (
+								<EmptySiteIcon>
+									<Gridicon icon="globe" />
+								</EmptySiteIcon>
+							) }
 						</FillDefaultIconWhite>
 					),
 				};
 			},
-		[ currentRoute, dispatch ]
+		[ currentRoute ]
 	);
 
 	return siteToAction;
 };
 
 export const useCommandPalette = ( {
+	currentSiteId,
 	selectedCommandName,
 	setSelectedCommandName,
 	search,
+	navigate,
+	useCommands,
+	wpcom,
+	currentRoute,
 }: useCommandPaletteOptions ): {
 	commands: Command[];
 	filterNotice: string | undefined;
 	emptyListNotice: string | undefined;
 } => {
-	const { data: allSites = [] } = useSiteExcerptsQuery(
-		[],
-		( site ) => ! site.options?.is_domain_only
-	);
-	const dispatch = useDispatch();
-	const siteToAction = useSiteToAction();
+	const { data: allSites = [] } = useSites( wpcom );
```

Comment:
This should run only on multi-site contexts. If the command palette is restricted to show single-site commands only, then we don't need to fetch all the sites.

Reply:
We can maybe think about this in the PR that adds the app, and keep this PR only focused on decoupling things from the Calypso state and making sure there are no regressions in Calypso.